log4j-api/src/test/java/org/apache/logging/log4j/util/LoaderUtilTest.java

```diff
@@ -55,4 +62,52 @@ public void systemClassLoader() {
             thread.setContextClassLoader(tccl);
         }
     }
+
+    @Test
+    public void securityManager() throws Exception {
+        String separator = System.getProperty("path.separator");
+        List<URL> classpath = new ArrayList<>();
+        for (String entry : System.getProperty("java.class.path").split(separator)) {
+            classpath.add(new File(entry).toURL());
+        }
+
+        URLClassLoader classLoader = new URLClassLoader(classpath.toArray(new URL[0]), null);
+        Class<?> thisClass = classLoader.loadClass("org.apache.logging.log4j.util.LoaderUtilTest");
+        Method getClassLoaders = thisClass.getMethod("runUnderSecurityManager");
+        getClassLoaders.invoke(null);
+        // no exception thrown
```

Comment:
You can use `assertDoesNotThrow(() -> ...);` to be more expressive about that than a comment.

Reply:
Neat! This does feel more satisfying (and surely yields a better error message too).